rancher/rancher-ui/infrastructure/hosts/custom/index.md

```diff
@@ -40,9 +40,9 @@ http://<rancher-server-ip>:8080/v1/projects/1a5/scripts/<registrationToken>
 
 ### Security Groups/Firewalls 
 
-For any hosts that are added, ensure that any security groups or firewalls allow traffic. If these are not enabled, the Rancher functionality will be limited.
+For any hosts that are added, ensure that any security groups or firewalls allow traffic between the Rancher server and the Rancher agents, otherwise Rancher functionality will be limited.
 
-* From and to all other hosts on UDP ports `500` and `4500` (for IPsec networking)
+* From Rancher server to UDP ports `500` and `4500` (for IPsec networking) on Rancher agent hosts
 
```

Comment:
Oh I see by "each other" Denise meant between the agents only. So given your example, the rules on host B would be:

```
ufw allow from C proto udp to B port 500

ufw allow from C proto udp to B port 4500

ufw allow from D proto udp to B port 500

ufw allow from D proto udp to B port 4500
```